tracee-rules/input.go

```diff
@@ -4,47 +4,49 @@ import (
 	"bufio"
 	"encoding/gob"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"log"
 	"os"
+	"strings"
 	"time"
 
 	"github.com/aquasecurity/tracee/tracee-rules/types"
 	tracee "github.com/aquasecurity/tracee/tracee/external"
 )
 
-func setupStdinSource(inputSource string) (chan types.Event, error) {
-	res := make(chan types.Event)
-	scanner := bufio.NewScanner(os.Stdin)
-	go func() {
-		for scanner.Scan() {
-			event := scanner.Bytes()
-			switch inputSource {
-			case "tracee":
-				var e tracee.Event
-				err := json.Unmarshal(event, &e)
-				if err != nil {
-					log.Printf("invalid json in %s: %v", string(event), err)
-				}
-				res <- types.Event(e)
-			}
-		}
-		close(res)
-	}()
-	return res, nil
+var helpErr = errors.New("help me if you can im feeling down")
+
+type inputFormat uint8
+
+const (
+	invalidInputFormat inputFormat = iota
+	jsonInputFormat
+	gobInputFormat
+)
+
+type traceeInputOptions struct {
+	inputFile   *os.File
+	inputFormat inputFormat
+	closeOnEOF  bool
 }
 
-func setupTraceeSource(traceeFilePath string) (chan types.Event, error) {
-	_, err := os.Stat(traceeFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("invalid Tracee input file: %s", traceeFilePath)
+func setupInputSource(opts *traceeInputOptions) (chan types.Event, error) {
```

Comment:
Is there something specific to tracee in the opts struct?
If not, better changing the name to `inputOptions`, especially if you change `setupTraceeSource`->`setupInputSource`

Reply:
This is a fair point, there's of course value in keeping code more 'general', but my philosophy is that we shouldn't complicate code for the sake of flexibility that we don't need yet. This isn't such a large chunk of code, when we support more input sources we can easily go back and refactor a bit.

Reply:
Ok, got it.
In that case, I think the name should be changed to `setupTraceeInputSource` as I mentioned above

Reply:
if you do that then same for setup{GOB,JSON}InputSource (add tracee since they accept traceeOptions)

Reply:
@grantseltzer this (and the others) still need to be changed

Reply:
Got it. Agreed they need to be changed but this will require a bit more work because  when `parseTraceeInputOptions` is called, the channel of type `chan tracee.Events` can't be assigned to `chan types.Events` (channels of types are themselves their own types). I'm not sure fixing this should be in this PR, I can fix that in a follow up if you'd like.

Reply:
I don't understand what you wrote :) we are just suggesting the rename the function name, shouldn't have any implications. but if you want to leave it, that's fine

Reply:
Lol I thought you were responding to https://github.com/aquasecurity/tracee/pull/500#issuecomment-772813353